src/debug.c

```diff
@@ -441,25 +459,40 @@ coap_show_pdu(const coap_pdu_t *pdu) {
   int content_format = -1;
   size_t data_len;
   unsigned char *data;
+#if defined(WITH_CONTIKI) || defined(WITH_LWIP)
```

Comment:
I would like to make these numbers depend on pre-processor symbols. Such as, e.g.,

```c
#ifndef COAP_DEBUG_BUF_SIZE
#if defined(WITH_CONTIKI) || defined(WITH_LWIP)
#define COAP_DEBUG_BUF_SIZE 128
#else /* defined(WITH_CONTIKI) || defined(WITH_LWIP) */
#define COAP_DEBUG_BUF_SIZE (8 + (COAP_DEFAULT_MAX_PDU_RX_SIZE) * 2)
#endif /* defined(WITH_CONTIKI) || defined(WITH_LWIP) */
#endif /* COAP_DEBUG_BUF_SIZE */
```
COAP_DEFAULT_MTU

Reply:
Will make the change - makes good sense

Reply:
Ouch - COAP_DEFAULT_MAX_PDU_RX_SIZE is over 8M bytes - which is too large for my heap.

I prefer that the max data size is 1024 as per RFC7252 4.6.  Message Size base don a normal MTU, but apps can change the the size with -DCOAP_DEBUG_BUF_SIZE